### PR TITLE
Lecture 27: Scalable Algorithms and Systems for Learning, Inference, and Prediction

### PR DESCRIPTION
**Pull request by SydsnapMisato**
rendered link: https://justhalf.github.io/pgm-spring-2019/notes/lecture-27/

### CODE REVIEW

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
- Solution 1: Bounded-Asynchronous Parallel (BAP)
```

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]: Please format this section in paragraph form so that it reads more like a blog post, rather than bullet points from lecture slides. Here's an example that I liked: https://sailinglab.github.io/pgm-spring-2019/notes/lecture-13/

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md:104]: Please format this section in paragraph form so that it reads more like a blog post, rather than bullet points from lecture slides. Here's an example that I liked: https://sailinglab.github.io/pgm-spring-2019/notes/lecture-13/

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md:156]: Please format this section in paragraph form so that it reads more like a blog post, rather than bullet points from lecture slides. Here's an example that I liked: https://sailinglab.github.io/pgm-spring-2019/notes/lecture-13/

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md:171]: Please format this section in paragraph form so that it reads more like a blog post, rather than bullet points from lecture slides. Here's an example that I liked: https://sailinglab.github.io/pgm-spring-2019/notes/lecture-13/

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
STRADS (STRucture-Aware Dynamic Scheduler) is a scheduler that allows dynamic scheduling of parameter updates in coordinate descent.
```

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
It focuses on two things:
```

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
- Dependency checking: If two coordinates are correlated, do not compute them in parallel.
```

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
- Prioritization: Choose the coordinate to update with a probability proportional to its historic rate of change. Intuitively, a parameter that changes rapidly will probably cause a quicker decrease of the loss function.
```

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md:227]: Please format this section in paragraph form so that it reads more like a blog post, rather than bullet points from lecture slides. Here's an example that I liked: https://sailinglab.github.io/pgm-spring-2019/notes/lecture-13/

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
The algorithm is summarized below:
```

File: _posts/2019-04-24-lecture-27.md
Comment (RLAgent) [inline _posts/2019-04-24-lecture-27.md]:
```suggestion
1. Split documents into $$P$$ partitions, with their corresponding topic mixture $$\delta_{di}$$.
```
For consistent style, place a period "." at the end, here and elsewhere.

**Review by RLAgent — CHANGES_REQUESTED** [review]
The notes look really good. I made a few suggestions: https://github.com/sailinglab/pgm-spring-2019/pull/35

Most importantly, please format the following sections in paragraph form so that it reads more like a blog post, rather than bullet points from lecture slides. Here's an example that I liked: https://sailinglab.github.io/pgm-spring-2019/notes/lecture-13/
- Parallel Stochastic Gradient Descent (PSGD)
- Shotgun, a Parallel Coordinate Descent Algorithm
- Block-greedy Coordinate Descent
- Accelerated Proximal Gradients
- Distributed Systems for ML

File: _posts/2019-04-24-lecture-27.md
Comment (justhalf) [reply _posts/2019-04-24-lecture-27.md]: @lileee Is this comment on line 387 on "Distributed Systems for ML" or the previous section? I feel the previous section is already in paragraph format.

**Comment by alshedivat** [comment]
@justhalf, @many-facedgod, @SydsnapMisato, @lileee, what's the status on these notes? Could you please iterate and merge by EOD today? Thanks.

**Comment by many-facedgod** [comment]
> @justhalf, @many-facedgod, @SydsnapMisato, @lileee, what's the status on these notes? Could you please iterate and merge by EOD today? Thanks.

We made the changes a few days ago. Waiting for a review from @lileee .

File: _posts/2019-04-24-lecture-27.md
Comment (alshedivat) [inline _posts/2019-04-24-lecture-27.md]: Could you please wrap this and other images with `<figure></figure>` tags as specified in [the template](https://github.com/sailinglab/pgm-spring-2019/blob/master/_posts/2019-01-09-lecture-notes-template.md)?

File: _posts/2019-04-24-lecture-27.md
Comment (alshedivat) [inline _posts/2019-04-24-lecture-27.md]: Same here: please use `<figure></figure>`.

File: _posts/2019-04-24-lecture-27.md
Comment (alshedivat) [inline _posts/2019-04-24-lecture-27.md]: Same here: please use `<figure></figure>`.

File: _posts/2019-04-24-lecture-27.md
Comment (alshedivat) [inline _posts/2019-04-24-lecture-27.md]: Same here: please use `<figure></figure>`.

File: _posts/2019-04-24-lecture-27.md
Comment (alshedivat) [inline _posts/2019-04-24-lecture-27.md]: Same here: please use `<figure></figure>`.

File: _posts/2019-04-24-lecture-27.md
Comment (alshedivat) [inline _posts/2019-04-24-lecture-27.md]: Same here: please use `<figure></figure>`.

**Review by alshedivat — CHANGES_REQUESTED** [review]
Please address some minor comments it'll be ready to merge. Thanks.

**Comment by alshedivat** [comment]
@many-facedgod, there are a few unresolved comments form @lileee. Please address those and the ones about figures I just added. Thanks.

**Comment by justhalf** [comment]
@alshedivat All suggestions made! Ready to merge?

**Comment by alshedivat** [comment]
@justhalf, almost there. Could you make sure figures are added exactly as given in the template: https://raw.githubusercontent.com/sailinglab/pgm-spring-2019/master/_posts/2019-01-09-lecture-notes-template.md

**Comment by many-facedgod** [comment]
@alshedivat  The image format has been fixed by @justhalf. Fine to merge now?

**Comment by alshedivat** [comment]
Great, thanks for all the fixes!

**Review by alshedivat — APPROVED** [review]

**Pull request merged**
